### PR TITLE
New parser, suitable for all versions.

### PR DESCRIPTION
**Pull request by gbm19**
Works for all files I tried. 
Tested on the files at http://users.physik.fu-berlin.de/~glaubitz/qtiplot/

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+# c++11 standard
+add_compile_options(-std=c++11)
+
 # boost library headers
 find_package(Boost REQUIRED)
 include_directories(${Boost_INCLUDE_DIRS})
 
 # library version
-set(LIBORIGIN_VERSION_MAJOR 2)
+set(LIBORIGIN_VERSION_MAJOR 3)
```
Comment (Saluev) [inline CMakeLists.txt]: So we are going ahead of the originally wrapped library now :)

File: OriginFile.cpp [hunk 2]
```diff
@@ -28,121 +28,105 @@
...
 
-	string vers(4, 0);
-	file.seekg(0x7, ios_base::beg);
-	file >> vers;
-	fileVersion = atoi(vers.c_str());
+	string vers;
+	getline(file, vers);
+	unsigned int majorVersion = strtol(vers.substr(5,1).c_str(),0,10);
+	char locale_decpoint = vers[6];
+	(void) locale_decpoint; // supress compiler warning
```
Comment (Saluev) [inline OriginFile.cpp]: Do you mean the unused variable warning?

Comment (gbm19) [inline OriginFile.cpp]: Yes

**Comment by gbm19** [comment]
Yes. I submitted a parallel (without callback stuff) pull request on the sourceforge liborigin code.
Also, I am working on a similar PR for scidavis.

File: OriginAnyParser.cpp [hunk 3]
```diff
@@ -0,0 +1,2820 @@
...
+			stmp.str(cvehd.substr(0x4A));
+			GET_SHORT(stmp, width)
+		}
+		int col_index = findColumnByName(ispread, name);
+		if (col_index != -1) {
+			if (speadSheets[ispread].columns[col_index].name != name)
+				speadSheets[ispread].columns[col_index].name = name;
+
+			SpreadColumn::ColumnType type;
+			switch(c){
```
Comment (Saluev) [inline OriginAnyParser.cpp]: This doesn't have `case 1`, is it intended?

Comment (gbm19) [inline OriginAnyParser.cpp]: It is the same code as in previous Origin750Parser.cpp

**Pull request merged**